schema/validator.go

```diff
@@ -38,7 +41,7 @@ func (e ValidationError) Error() string {
 }
 
 // Validate validates the given reader against the schema of the wrapped media type.
-func (v Validator) Validate(src io.Reader) error {
+func (v Validator) Validate(src io.Reader, verbose bool, out *log.Logger) error {
```

Comment:
@stevvooe 
I'm blocked is that validate functions (ValidateLayout/Validate/(Validator).Validate) (maybe) MUST return new variable of digest list, for these functions calculated digests internally. 
If digest list can be added to validate functions when returning, I can utilize them out of validate-function. Otherwise, spaghetti code seems can't work out the problem because of these internal digests variable.
new returning variable is needed.